src/views/Correction.vue

```diff
@@ -331,20 +330,20 @@ export default {
     }
 
     // this is the id of THIS correction filing
-    // if falsy, this is a new correction filing
+    // if 0, this is a new correction filing
     // otherwise it's a draft correction filing
-    this.filingId = +this.$route.params.id // number (may be NaN)
+    this.filingId = +this.$route.params.id || 0 // number
 
     // this is the id of the original filing to correct
     this.correctedFilingId = +this.$route.params.correctedFilingId // number (may be NaN)
 
     // if required data isn't set, route to home
-    if (!this.entityIncNo || !this.correctedFilingId) {
+    if (!this.entityIncNo || isNaN(this.correctedFilingId)) {
```

Comment:
Was `!this.correctedFilingId)` still returning true in some instances because it was a string, albeit sometimes an empty one?